src/libraries/System.Net.Quic/src/System/Net/Quic/QuicConnection.cs

```diff
@@ -588,6 +633,15 @@ private unsafe int HandleEventPeerStreamStarted(ref PEER_STREAM_STARTED_DATA dat
         data.Flags |= QUIC_STREAM_OPEN_FLAGS.DELAY_ID_FC_UPDATES;
         return QUIC_STATUS_SUCCESS;
     }
+    private unsafe int HandleEventStreamsAvailable(ref STREAMS_AVAILABLE_DATA data)
+    {
+        int bidirectionalStreamsCountIncrement = data.BidirectionalCount - _availableBidirectionalStreamsCount;
+        int unidirectionalStreamsCountIncrement = data.UnidirectionalCount - _availableUnidirectionalStreamsCount;
+        Volatile.Write(ref _availableBidirectionalStreamsCount, data.BidirectionalCount);
+        Volatile.Write(ref _availableUnidirectionalStreamsCount, data.UnidirectionalCount);
+        OnStreamsAvailable(bidirectionalStreamsCountIncrement, unidirectionalStreamsCountIncrement);
+        return QUIC_STATUS_SUCCESS;
+    }
```

Comment:
We receive events only when the available streams count increases, right? If so, isn't there a race between decrementing `_available*directionalStreamsCount` from `OpenOutboundStreamAsync` and processing this event?

Reply:
You're concerned about the increments not being accurate?

Reply:
both the increments and `Available*StreamsCount` properties. We have no idea if the stream we may be concurrently creating is counted in the numbers reported from MsQuic. This may lead us to trying to send a request on a connection where opening the request stream will block, negating the intended gain from those changes.

I think we need to ask MsQuic to include either the deltas or the absolute numbers in the event.

Reply:
This has been fixed, the properties with counts are removed, remains only private fields that are manipulated only from MsQuic thread which processes events synchronously for a connection.